src/metrics/perf_event.rs

```diff
@@ -5,6 +5,8 @@ use std::fmt;
 
 #[derive(Deserialize, Clone, Debug)]
 pub enum HardwareCounter {
+    /// The type of hardware metric one wants to record.This is done while building the application and adding HardwareCounter
+    /// enum options to it.
```

Comment:
What is there now is a rustdoc comment for the first Enum variant. It needs to go on top.

```suggestion
/// An enum representing supported hardware counters
///
/// It is a wrapper around [Hardware] in order to support [Deserialize]
#[derive(Deserialize, Clone, Debug)]
pub enum HardwareCounter {

```